Assets/Scripts/InputManager.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class InputManager : MonoBehaviour
+{
+    private static InputManager _instance;
+    public List<SourceBuildingAsset> buildingList = new List<SourceBuildingAsset>();
+    public UnityEvent CreateBuilding = new UnityEvent();
+    public UnityEvent DestroyBuilding = new UnityEvent();
+    public UnityEvent UpgradeBuilding = new UnityEvent();
+    public UnityEvent RotateBuilding = new UnityEvent();
+    public UnityEvent<SourceBuildingAsset> ChangeBuilding = new UnityEvent<SourceBuildingAsset>();
+
+    public Dictionary<string, SourceBuildingAsset> buildings;
+
+    public static InputManager Instance
+    {
+        get
+        { 
+            return _instance; 
+        }
+    }
+    
+    private void Awake()
+    {
+        if (_instance != null && _instance != this) 
+        {
+            Destroy(this.gameObject);
+        }
+ 
+        _instance = this;
+        DontDestroyOnLoad( this.gameObject );
+    }
+
+    private void Start()
+    {
+        buildings = new Dictionary<string, SourceBuildingAsset>() 
```

Comment:
maybe you can use the enum core building types here instead of strings?

Reply:
we need to resolve what is going on with all the enums and which one to use and streamline sourcebuilding asset and productionbuildingasset into one thing

Reply:
alright we'll do that